sdv/metadata/single_table.py

```diff
@@ -245,11 +244,90 @@ def to_dict(self):
 
         return deepcopy(metadata)
 
+    def _determine_sdtype_for_numbers(self, data):
+        """Determine the sdtype for a numerical column.
+
+        Args:
+            data (pandas.Series):
+                The data to be analyzed.
+        """
+        sdtype = 'numerical'
+        if len(data) > 5:
+            whole_values = (data == data.round()).loc[~data.isna()].all()
+            positive_values = (data >= 0).loc[~data.isna()].all()
+
+            unique_values = data.nunique()
+            percentage_unique_values = unique_values <= round(len(data) / 10)
+
+            if whole_values and positive_values and percentage_unique_values:
+                sdtype = 'categorical'
+            elif unique_values == len(data) and whole_values:
+                sdtype = 'id'
+
+        return sdtype
+
+    def _determine_sdtype_for_objects(self, data):
+        """Determine the sdtype for an object column.
+
+        Args:
+            data (pandas.Series):
+                The data to be analyzed.
+        """
+        sdtype = None
+        try:
+            datetime_format = get_datetime_format(data)
+            pd.to_datetime(data, format=datetime_format, errors='raise')
```

Comment:
I thought that we need to ensure here that all the data is convertible with the same datetime format. Should we subsample? I didn't look at the computational time.

Reply:
The goal is to check that everything matches, but since it can possibly be slow for very large datasets, we can probably pick a max number of rows and say if at least that many match it's probably fine